### PR TITLE
fix(53754): Re-exported symbol marked with deprecated doesn't get correct deprecation highlighting

### LINKED ISSUE #53754: Re-exported symbol marked with `deprecated` doesn't get correct deprecation highlighting
Type: <b>Possible Bug</b>

In the following example vscode does not seem to pick up the @deprecated tag and I do not understand why? The use case is to mark the lodash template function as deprecated.

```typescript
import _template from 'lodash/template';

/** @deprecated Deprecated. Use JavaScript template string instead. */
export {_template as template};
```

VS Code version: Code 1.77.2 (e344f1f539a80912a0e9357cec841f36ce97a4e2, 2023-04-06T10:12:31.126Z)
OS version: Darwin x64 22.4.0
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz (8 x 2700)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|4, 10, 7|
|Memory (System)|16.00GB (0.03GB free)|
|Process Argv|--crash-reporter-id 632d5250-6707-4e87-8d1c-3e7bcfc44bfe|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (20)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|2.4.0
prettier-vscode|esb|9.10.4
svn-scm|joh|2.15.7
vscode-docker|ms-|1.24.0
vscode-dotnet-runtime|ms-|1.6.0
playwright|ms-|1.0.10
isort|ms-|2022.8.0
python|ms-|2023.6.0
vscode-pylance|ms-|2023.4.10
remote-containers|ms-|0.288.0
hexeditor|ms-|1.9.11
oracledevtools|Ora|21.5.0
refactor|p42|3.0.1
vscode-print|pdc|0.11.1
quicktype|qui|12.0.46
LiveServer|rit|5.7.9
vscode-styled-components|sty|1.7.8
vscode-stylelint|sty|1.2.4
vscode-icons|vsc|12.3.0
vscode-todo-highlight|way|1.0.5

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
nodejswelcome1:30587005
282f8724:30602487
pyind779:30671433
f6dab269:30613381
vscrp:30673768
pythonsymbol12:30671437
a9j8j154:30646983
vsctsb:30705553
defaultazurewalk:30687958
pythonms35:30701012
pythonfmttextcf:30710682
pythonclientmv:30708685

```

</details>

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #53754

### CODE REVIEW

**Review by sandersn — CHANGES_REQUESTED** [review]
Looks good, just one small formatting suggestion

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -44123,19 +44127,17 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
...
                 }
             }
         }
     }
 
-    function isDeprecatedAliasedSymbol(symbol: Symbol) {
-        return !!symbol.declarations && every(symbol.declarations, d => !!(getCombinedNodeFlags(d) & NodeFlags.Deprecated));
-    }
-
     function checkDeprecatedAliasedSymbol(symbol: Symbol, location: Node) {
```
[the remark this answers is not in the document]
Comment (a-tarasyuk) [reply src/compiler/checker.ts:44137]: What would be a better name for this function?

[the remark this answers is not in the document]
Comment (sandersn) [reply src/compiler/checker.ts:44137]: It's primarily a getter now, so maybe `getFirstDeprecatedDeclarationOfAlias` or `findFirstDeprecatedDeclarationOfAlias`? The error checking is incidental; it only happens for some outputs.

[the remark this answers is not in the document]
Comment (sandersn) [reply src/compiler/checker.ts:44137]: I like that name. Want to change it, and then I'll merge this?

**Review by sandersn — APPROVED** [review]

**Pull request merged**